File: 198. House Robber/198. House Robber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function (nums) {
  const n = nums.length;
  if (n == 1) {
    return nums[0];
  }

  let rob1 = nums[0];
  let rob2 = Math.max(rob1, nums[1]);

  for (let i = 2; i < n; i++) {
    [rob1, rob2] = [rob2, Math.max(rob2, rob1 + nums[i])];
  }

  return rob2;
};

File: 198. House Robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]

        prev2 = 0
        prev1 = 0

        for n in nums:
            not_rob = prev1
            rob = prev2 + n
            prev2, prev1 = prev1, max(not_rob, rob)

        return prev1
